notification-service/pom.xml

```diff
@@ -24,6 +24,17 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.java.mentorship</groupId>
+            <artifactId>notification-contracts</artifactId>
+            <version>1.0-SNAPSHOT</version>
```

Comment:
Let's declare this version in the parent pom in dependencyManagement

Also, internal dependencies are generally declared first, for visibility